service/src/main/java/bio/terra/workspace/service/workspace/flight/WorkspaceDeleteFlight.java

```diff
@@ -25,22 +26,30 @@ public WorkspaceDeleteFlight(FlightMap inputParameters, Object applicationContex
     WorkspaceStage workspaceStage =
         WorkspaceStage.valueOf(
             inputParameters.get(WorkspaceFlightMapKeys.WORKSPACE_STAGE, String.class));
+    UUID workspaceId =
+        UUID.fromString(inputParameters.get(WorkspaceFlightMapKeys.WORKSPACE_ID, String.class));
     // TODO: we still need the following steps once their features are supported:
-    // 1. delete controlled resources using the Cloud Resource Manager library
+    // 1. Delete the cloud contexts from non-GCP cloud platforms
     // 2. Notify all registered applications of deletion, once applications are supported
     // 3. Delete policy objects in Policy Manager, once it exists.
 
     RetryRule retryRule =
         new RetryRuleExponentialBackoff(
             INITIAL_INTERVALS_SECONDS, MAX_INTERVAL_SECONDS, MAX_OPERATION_TIME_SECONDS);
 
+    addStep(
+        new ListControlledResourceIdsStep(appContext.getResourceDao(), workspaceId, null),
```

Comment:
in which step do we delete the controlled resources from the database?

Reply:
clarified in comment but we never explicitly do this when deleting a workspace, we have resources set up to cascade on workspace deletion in the DB.